src/Authenticator/AbstractAuthenticator.php

```diff
@@ -78,6 +85,32 @@ public function setIdentifier(IdentifierInterface $identifier)
         return $this;
     }
 
+    /**
+     * Get the last authentication result instance stored or null if none.
+     *
+     * @return ResultInterface|null
+     */
+    public function getLastResult()
+    {
+        return $this->_lastResult;
+    }
+
+    /**
+     * Build the result and set as last
+     *
+     * @param null|array|\ArrayAccess $data The identity data
+     * @param string $status Status constant equivalent.
+     * @param array $messages Messages.
+     * @throws \InvalidArgumentException When invalid identity data is passed.
+     * @return ResultInterface
+     */
+    protected function _buildLastResult($data, $status, array $messages = [])
```

Comment:
having this method does not guarantee that all custom class should use this method in their class

Reply:
any suggestions of how force the use?

Reply:
I'm not sure 
but maybe you able to  store it in `result` class

Reply:
Since we have `getLastResult()` method maybe this should be just a setter (setLastResult), not a combined factory method/setter?